src/diff.c

```diff
@@ -178,7 +178,7 @@ diff_common_add_diff_stat(struct view *view, const char *text, size_t offset)
 
 	/* Ensure that '|' is present and the file name part contains
 	 * non-space characters. */
-	if (!pipe || pipe == data || strcspn(data, " ") == 0)
+	if (!pipe || pipe == data || strlen(pipe) == len)
```

Comment:
I agree that it's hard to follow and will revise.

This doesn't fix a false positive, but was an attempt to make the code match the existing comment.

I noticed that the clause `strcspn(data, " ") == 0` can't ever be true, since we already scanned past leading space when initializing `char *data = start + strspn(start, " ")`.